### PR TITLE
Fixed #32780 -- Made Add/RemoveConstraint operations a noop for covering/deferrable unique constraints on SQLite.

### PR DESCRIPTION
**Pull request by hannseman**
Ticket: https://code.djangoproject.com/ticket/32780

### CODE REVIEW

**Comment by hannseman** [comment]
Realised that we have the a similar issue with `UniqueConstraint.opclasses` but in that case we don't have a system check or even a database feature flag. Don't know if that's intended but if not I can add a system check and a feature-flag and use it to exclude `opclasses` for non PostgreSQL backends.

File: tests/schema/tests.py [hunk 1]
```diff
@@ -2438,6 +2438,16 @@ def test_func_unique_constraint_nondeterministic(self):
             with self.assertRaises(DatabaseError):
                 editor.add_constraint(Author, constraint)
 
+    @skipIfDBFeature('supports_covering_indexes')
+    def test_unique_constraint_covering_unsupported(self):
```
Comment (felixxm) [inline tests/schema/tests.py:2442]:
```suggestion
    def test_covering_unique_constraint_noop(self):
```

**Review by felixxm — COMMENTED** [review]
@hannseman Thanks :+1: Please do the same for deferrable unique constraints.

> Realised that we have the a similar issue with `UniqueConstraint.opclasses` but in that case we don't have a system check or even a database feature flag. Don't know if that's intended but if not I can add a system check and a feature-flag and use it to exclude `opclasses` for non PostgreSQL backends.

It's documented that `opclasses` is ignored for databases besides PostgreSQL, so a constraint will be created but without opclasses. I think it's expected and an extra check or a feature flag is not needed. However there is an issue on SQLite where it's completely ignored so we should add the `constraint.opclasses` check to the `add_constraint()`/`remove_constraint()` on SQLite:
```diff
diff --git a/django/db/backends/sqlite3/schema.py b/django/db/backends/sqlite3/schema.py
index b089a9d355..a761a71861 100644
--- a/django/db/backends/sqlite3/schema.py
+++ b/django/db/backends/sqlite3/schema.py
@@ -420,7 +420,7 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
 
     def add_constraint(self, model, constraint):
         if isinstance(constraint, UniqueConstraint) and (
-            constraint.condition or constraint.contains_expressions
+            constraint.condition or constraint.contains_expressions or constraint.opclasses
         ):
             super().add_constraint(model, constraint)
         else:
@@ -428,7 +428,7 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
 
     def remove_constraint(self, model, constraint):
         if isinstance(constraint, UniqueConstraint) and (
-            constraint.condition or constraint.contains_expressions
+            constraint.condition or constraint.contains_expressions or constraint.opclasses
         ):
             super().remove_constraint(model, constraint)
         else:
```
Please do this in a separate PR (and ticket).

**Comment by felixxm** [comment]
@hannseman Thanks for updates :+1:

**Pull request merged**